python/python/bystro/utils/compress.py

```diff
@@ -31,24 +31,61 @@ def _get_gzip_program_path() -> str:
 IS_BGZIP = GZIP_EXECUTABLE.endswith("bgzip")
 
 
-def get_compress_from_pipe_cmd(out: str) -> str:
+def get_compress_from_pipe_cmd(
+    out: str, pipe_from_cmd: str | None = None, pipe_to_cmd: str | None = None
+) -> str:
+    """
+    Generate a command string for compressing data from a pipe.
+
+    Args:
+        out (str): The output path for the compressed data.
+        pipe_from_cmd (str | None, optional): The command to pipe data from. Defaults to None.
+        pipe_to_cmd (str | None, optional): The command to pipe data to. Defaults to None.
+
+    Returns:
+        str: The command string for compressing data from a pipe.
+    """
+    cmd = None
     if (IS_PIGZ or IS_BGZIP or IS_GZIP) and not out.endswith(".gz"):
         raise ValueError("Output path must end with .gz")
 
     if IS_PIGZ:
-        return f"{GZIP_EXECUTABLE} -c -p {NUM_CPUS} > {out}"
+        cmd = f"{GZIP_EXECUTABLE} -c -p {NUM_CPUS}"
+    elif IS_BGZIP:
+        cmd = f"{GZIP_EXECUTABLE} -c --index --index-name {out}.gzi --threads {NUM_CPUS}"
+    else:
+        cmd = f"{GZIP_EXECUTABLE} -c"
 
-    if IS_BGZIP:
-        return f"{GZIP_EXECUTABLE} -c --index --index-name {out}.gzi --threads {NUM_CPUS} > {out}"
+    if pipe_from_cmd:
+        cmd = f"{pipe_from_cmd} | {cmd}"
 
-    return f"{GZIP_EXECUTABLE} -c > {out}"
+    if pipe_to_cmd:
+        cmd = f"{cmd} | {pipe_to_cmd}"
 
+    return f"{cmd} > {out}"
 
-def get_decompress_to_pipe_cmd(input_path: str) -> str:
-    if IS_PIGZ:
-        # pigz doesn't benefit much from more than 2 threads during decompression
-        return f"{GZIP_EXECUTABLE} -d -c {input_path} -p 2"
+
+def get_decompress_to_pipe_cmd(input_path: str, pipe_to_cmd: str | None = None) -> str:
```

Comment:
updated to support bystro-vcf in a single pipe command (e.g. bgzip -d -c file.gz | bystro-vcf)